src/compile.ts

```diff
@@ -45,6 +46,12 @@ export function compile(
   return (ctx) => {
     const functionless = ts.factory.createUniqueName("functionless");
     return (sf) => {
+      // Do not transform any of the files matched by "exclude"
+      if (excludeMatchers.some((matcher) => matcher.test(sf.fileName))) {
+        return sf;
+      }
+
+      // TODO: this seems like a pattern, make a "public import context"
```

Comment:
I changed compile to only include the functionless import when its used, instead of all of the time. 

Originally thought we'd have multiple dynamic imports, but ended up removing the other one. Removed the comment.